docs/src/main/asciidoc/extension-authors-guide.adoc

```diff
@@ -494,15 +494,15 @@ public class FileConfig {
     String format;
 
     /**
-     * The file log level.
+     * The log level.
      */
     @ConfigItem(defaultValue = "ALL")
     Level level;
 
     /**
-     * The file logging log level.
+     * The file name where logs will be stored.
      */
-    @ConfigItem(defaultValue = "quarkus.log")
+    @ConfigItem(defaultValue = "application.log")
```

Comment:
This change should be accompanied by an update to `.gitignore` to change `quarkus.log` to `application.log`, else we'll see it show up accidentally in commits before long.

Reply:
I also see a reference to `quarkus.log` in `test-framework/common/src/main/java/io/quarkus/test/common/PropertyTestUtil.java`.  Better make sure that won't be affected.

Reply:
Should we really change the default log filename? I don't see the new one as an improvement.

Let's remove this change from this PR, it should be discussed separately as mentioned by @dmlloyd .

Reply:
Hi @dmlloyd, @gsmet. I do not agree. Please note that this modification does not change any logic linked to logging, nor default logs placement nor any file names connected to logging mechanisms. This change is embedded in context of this specific example, an example demonstrating how configuration can be structured into classes, and should be treated as such (as example, not as a reference where logs are located and in which files). I modified this line because file name used here, namely `quarkus.log`, is very confusing in regards to the matter of this example, where users need to understand how it coincides with the `quarkus.log` namespace (not file name) used in configuration. So not to confuse `quarkus.log` namespace with `quarkus.log` file I changed it to `application.log`.

Reply:
So what I would suggest to have your comments addressed and keep the example simple and straightforward, is to change `FileConfig.java` by adding:

```java
public static final String DEFAULT_LOG_FILE_NAME = "quarkus.log";
```

And use it there:

```java
@ConfigItem(defaultValue = DEFAULT_LOG_FILE_NAME)
File path;
```

And in `PropertyTestUtil.java` by changing:

```java
public static String getLogFileLocation() {
    return getLogFileLocation("quarkus.log");
}
```

To:

```java
public static String getLogFileLocation() {
    return getLogFileLocation(FileConfig.DEFAULT_LOG_FILE_NAME);
}
```

Same static reference will be used in example we are discussing and keep it non confusing. I hope this is ok.

Reply:
Please see https://github.com/quarkusio/quarkus/pull/2589/commits/5ed96c05b1e317e973a3a3eeea0ed44a4a596ed3

Reply:
Maybe using an existing system like logging in this doc was a mistake. :)

We'll always be trying to keep it in sync, and there will be questions when it isn't.

Reply:
Also it makes the PR difficult to review...

Reply:
@dmlloyd Yes, it would be better if we came up with a better example in the past (not an exact copy of internal logging system configuration), but since it's already there it would be a waste of time to implement something new. IMO this example is sufficient and we should simply not try to keep it in sync since this example is only to demonstrate how configuration can be composed into structure, not how logging can, or should be configured. Moreover, it's already out-of-sync because e.g. `RotationConfig` is missing in example, where it's used in the real system.

Reply:
All the more reason why it's a bad example.  And it's clearly not a waste of time to change it at all, because you would invest time and the result would be something clearer (forever).  Saying it's a waste of time to fix this is like saying that it's a waste of time to ever improve anything, so long as it meets a bare minimum standard.  It's OK to have high standards!